application/src/test/java/org/togetherjava/tjbot/commands/mediaonly/MediaOnlyChannelListenerTest.java

```diff
@@ -0,0 +1,87 @@
+import org.togetherjava.tjbot.jda.JdaTester;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+class MediaOnlyChannelListenerTest {
+
+    private JDA jda;
+    private final Message message = mock(Message.class);
+    private final long responseNumber = 1L;
+    private final User user = mock(User.class);
+    private MediaOnlyChannelListener mediaOnlyChannelListener;
+
+    @BeforeEach
+    void setUp() {
+        JdaTester jdaTester = new JdaTester();
+        jda = jdaTester.getJdaMock();
+
+        Config config = mock(Config.class);
+        when(config.getMediaOnlyChannelPattern()).thenReturn("memes");
+        mediaOnlyChannelListener = new MediaOnlyChannelListener(config);
+    }
+
+    @Test
+    void validMessagePostWithAttachment() {
+        Message.Attachment attachment = new Message.Attachment(1L, null, null, "TEST",
+                "Test ContentType", null, 1, 1, 1, false, (JDAImpl) jda);
+        List<Message.Attachment> attachments = Collections.singletonList(attachment);
+
+        when(message.getAuthor()).thenReturn(user);
+        when(message.getAttachments()).thenReturn(attachments);
+
+        MessageReceivedEvent messageReceivedEvent =
+                new MessageReceivedEvent(jda, responseNumber, message);
+        mediaOnlyChannelListener.onMessageReceived(messageReceivedEvent);
+        verify(message).getAttachments();
+    }
+
+    @Test
+    void validMessagePostWithUrlEmbedded() {
+        MessageEmbed messageEmbed = new MessageEmbed("https://9gag.com/gag/a61A238", "Test", "Test",
+                EmbedType.LINK, null, 1, null, null, null, null, null, null, null);
+        List<MessageEmbed> messageEmbeds = Collections.singletonList(messageEmbed);
+
+        when(message.getAuthor()).thenReturn(user);
+        when(message.getEmbeds()).thenReturn(messageEmbeds);
+
+        MessageReceivedEvent messageReceivedEvent =
+                new MessageReceivedEvent(jda, responseNumber, message);
+        mediaOnlyChannelListener.onMessageReceived(messageReceivedEvent);
+        verify(message).getEmbeds();
+    }
+
+    @Test
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    void unvalidMessagePostWithOnlyText() {
+        AuditableRestAction<Void> auditableRestAction = mock(AuditableRestAction.class);
+        RestAction restActionMessage = mock(RestAction.class);
+
+        when(message.getAuthor()).thenReturn(user);
+        when(message.getAttachments()).thenReturn(Collections.emptyList());
+        when(message.delete()).thenReturn(auditableRestAction);
+        when(auditableRestAction.flatMap(any())).thenReturn(restActionMessage);
+
+        MessageReceivedEvent messageReceivedEvent =
+                new MessageReceivedEvent(jda, responseNumber, message);
+        mediaOnlyChannelListener.onMessageReceived(messageReceivedEvent);
+        verify(message).getAttachments();
+    }
+}
```

Comment:
These unit tests dont play very well with the current test architecture.

The message event part should go into the helper:
```java
MessageReceivedEvent messageReceivedEvent = new MessageReceivedEvent(jda, responseNumber, message);
```
I would want to have a helper like `event = jdaHelper.createMessageEvent(message)` for that.

For the `User user = mock(User.class);` part, prefer `jdaTester.getMemberSpy()` or `jdaTester.createMemberSpy(123)` if you need a custom ID.

The attachment creation part looks ugly:
```java
Message.Attachment attachment = new Message.Attachment(1L, null, null, "TEST", "Test ContentType", null, 1, 1, 1, false, (JDAImpl) jda);```
Either mock the attachment or create an utility in the `jdaHelper` to create attachments. Maybe even with a builder, if needed.

There is some duplication in ur tests, for example they all create events and then trigger the onReceived thing etc. Might be worth to add a little private helper method in this class that does this, such as `sendAndReceiveMessage(message)`

Reply:
Could you please tell me, where is the test architecture described, i didnt see any in the other tests they are mostly random and follow no real architecture.

event = jdaHelper.createMessageEvent(message)  => if i should create this, pls describe exactly how you want this implemented...

Reply:
Well, the purpose is to move stuff that will be commonly used into the helper - so that everyone will benefit from it.

To summarize:
* the functionality to create a message event should be in `JdaTestHelper` (who is then responsible for using its `JdaImpl` and figuring out a `responseNumber`)
* you should use the existing `jdaTester.getMemberSpy/createMemberSpy` instead of creating ur own users
* attachment creation should be offered by `JdaTestHelper`
* u should reduce ur code duplication in ur test

Reply:
> As an example for how `JdaTestHelper` is useful, see how easy it makes to create slash events or button events. It has a rich API for that. The same idea should be applied to all other JDA topics.
> 
> If that is too much for you, that is fair. I can also take that part over.

i am fine implmeneting this (maybe i will ask a few questions later on...), but i think this could be extracted to an extra ticket this was never mention in the initial issue.  

This issue feels like a bit of feature creep and the initial feature was arleady done, but i get it why you want to add this to the  JdaTestHelper 😄

Reply:
Sure. But then I would vote for removing the unit test from this ticket and doing that in the other instead.

I just dont want to add "not so nice"-code to the project - from experience, it will start to rot then.